service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -151,9 +198,15 @@ func ReclaimResourcesWorkflow(ctx workflow.Context, params ReclaimResourcesParam
 	}
 
 	// Step 2. Sleep before deleting namespace from a database.
-	err = workflow.Sleep(ctx, params.NamespaceDeleteDelay)
-	if err != nil {
-		return result, fmt.Errorf("%w: %v", errors.ErrUnableToSleep, err)
+	for namespaceDeleteDelay > 0 {
```

Comment:
why `for`, and not `if`?

Reply:
I would also extract this into separate function.

Reply:
Because new timer needs to be created if delay is updated.